### DISCUSSION TITLE
Issue with Http2 Integration Understanding

### DISCUSSION THREAD

**Discussion by prabhatmishra33** [sindresorhus/got#1708] [General]
#### What would you like to discuss?

I was to discuss about the http2 implementation with GOT. Below changes i have done as to implement http2 for GOT.
- Upgraded Node to v15.10.0
- added `http2: true` into my options. you can see below
`		let options = {
			retry: httpClient.getRetryConfig(url),
			timeout: 5000,
			responseType: 'json',
			agent: httpClient.getHttpAgent(),
			maxContentLength: Infinity,
			http2: true
		};
`
- Added http2 in the agent
`
          const HttpAgent = require('agentkeepalive');
          const { HttpsAgent } = HttpAgent;
          const Http2Wrapper = require('http2-wrapper');
        
        http: new HttpAgent(),
        https: new HttpsAgent(),
        http2: new Http2Wrapper.Agent()
`
Please let me know if some more changes needed to be integrated 
...

#### Checklist
node v15.10.0
"got": "^11.8.2"
"http2-wrapper": "^1.0.3"
"agentkeepalive": "^4.1.4"

- [x] I have read the documentation.
https://www.npmjs.com/package/got#http2

**Comment by sindresorhus** [comment]
What issue are you seeing?

**Reply by prabhatmishra33** [reply]
Hi @sindresorhus  receiving `via: '1.1 nghttpx',` as 1.1 instead of `2`

**Reply by szmarczak** [reply]
It's just a header. I suppose the nghttpx proxy converts the request to 1.1 on server side and gives back in h2.

```js
curl --http2 https://nghttp2.org/httpbin/anything -v
```

**Reply by prabhatmishra33** [reply]
@sindresorhus @szmarczak  could you please tell me how can i achieve this using `GOT` ? 
As in response headers i still receive `1.1`
`{
  ':status': 200,
  date: 'Sat, 01 May 2021 17:43:00 GMT',
  'content-type': 'application/json',
  'content-length': '372',
  'access-control-allow-origin': '*',
  'access-control-allow-credentials': 'true',
  'x-backend-header-rtt': '0.003452',
  'strict-transport-security': 'max-age=31536000',
  server: 'nghttpx',
  via: '1.1 nghttpx',
  'alt-svc': 'h3-29=":443"; ma=3600',
  'x-frame-options': 'SAMEORIGIN',
  'x-xss-protection': '1; mode=block',
  'x-content-type-options': 'nosniff',
  [Symbol(nodejs.http2.sensitiveHeaders)]: []
}`

**Reply by szmarczak** [reply]
I don't know what's your problem. Everything works correctly here.

**Reply by prabhatmishra33** [reply]
Hi @szmarczak In you below example as per document
const got = require('got');

(async () => {
const {headers} = await got('https://nghttp2.org/httpbin/anything', {http2: true});
console.log(headers.via);
//=> '2 nghttpx'
})();

if via: '2 nghttpx then we consider this as an api using HTTP2 but in my case it's coming as via: '1.1 nghttpx'

**Reply by szmarczak** [reply]
Don't worry, the request is still via HTTP/2. Have you read https://github.com/sindresorhus/got/discussions/1708#discussioncomment-680554

**Reply by prabhatmishra33** [reply]
ok Thanks for confirming.
